### PR TITLE
AddProblemReducer emits expected app state

### PR DESCRIPTION
**Pull request by T-Davis**
AddProblemReducer emits expected app state
Part of [Error Handling UC 1](https://github.com/enspyrco/crowdleague/issues/274)

### CODE REVIEW

File: test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          GeneralProblem, 'Problem error message', StackTrace.current);
+
+      // Invokes the reducer to rebuild appState.
+      final newState =
+          reducer.reducer(initialState, AddProblem(problem: problem));
+
+      expect(newState.problems.first, problem);
+      expect(newState.pagesData.length, 2);
+      expect((newState.pagesData[1] as ProblemPageData).problem.message,
+          problem.message);
```
Comment (T-Davis) [inline test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart:31]: @nickmeinhold thanks for the awesome feedback!

File: lib/models/navigation/page_data/problem_page_data.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+import 'package:crowdleague/models/navigation/page_data/page_data.dart';
+import 'package:crowdleague/models/problems/problem_base.dart';
+import 'package:crowdleague/utils/serializers.dart';
+import 'package:meta/meta.dart';
+
+part 'problem_page_data.g.dart';
+
+abstract class ProblemPageData
```
Comment (nickmeinhold) [inline lib/models/navigation/page_data/problem_page_data.dart:14]: Would you be up for putting some clothes on this guy? :-)

File: lib/reducers/app_reducer.dart [hunk 3]
```diff
@@ -72,8 +73,11 @@ class StoreProcessingFailuresReducer
 
 class AddProblemReducer extends TypedReducer<AppState, AddProblem> {
   AddProblemReducer()
-      : super((state, action) =>
-            state.rebuild((b) => b..problems.add(action.problem)));
+      : super((state, action) => state.rebuild((b) => b
```
Comment (nickmeinhold) [inline lib/reducers/app_reducer.dart:76]: Cool, so we're adding a ProblemPageData with each new problem.  Nice place to start hey... looks good!

File: test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import 'package:crowdleague/actions/navigation/add_problem.dart';
```
Comment (nickmeinhold) [inline test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart:1]: Happy to change the file name to all snake case?

Comment (nickmeinhold) [reply test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart:1]: Test looks great!

File: test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart (hunk 1 above)
Comment (nickmeinhold) [reply test/sections/error-handling/UC1/reducer/addProblem_reducer_test.dart:31]: So welcome mate, that was fun!

**Review by nickmeinhold — CHANGES_REQUESTED** [review]
Looks great man, thanks!  Just a couple of tiny nits if you agree.

File: lib/models/navigation/page_data/problem_page_data.dart (hunk 2 above)
Comment (T-Davis) [reply lib/models/navigation/page_data/problem_page_data.dart:14]: Yep! I almost just pushed another commit to do just that. Workin'!

Comment (nickmeinhold) [reply lib/models/navigation/page_data/problem_page_data.dart:16]: Looks great!

**Review by nickmeinhold — CHANGES_REQUESTED** [review]
Hey mate, the file name change looks like it was missed... are you happy to change `test/sections/error-handling/UC1/reducer/addProblemReducer_test.dart` to `test/sections/error-handling/UC1/reducer/add_problem_reducer_test.dart`?   Just to keep everything in snake case

**Review by T-Davis — COMMENTED** [review]
> Hey mate, the file name change looks like it was missed

My bad! You asked for 🐍  and I gave 🐫 
I got it this time 🤓

**Review by nickmeinhold — APPROVED** [review]
Sorry for the delay, I didn't see that you'd made that last change and requested review... I think I turned email notifications off one day when I was sick, I've turned them back on now so I shouldn't miss these in future.

Awesome work on the PR mate!

**Pull request merged**